src/Squidex/Areas/Api/Controllers/Apps/AppPatternsController.cs

```diff
@@ -0,0 +1,131 @@
+﻿// ==========================================================================
+//  AppPatternsController.cs
+//  Squidex Headless CMS
+// ==========================================================================
+//  Copyright (c) Squidex Group
+//  All rights reserved.
+// ==========================================================================
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using NSwag.Annotations;
+using Squidex.Areas.Api.Controllers.UI.Models;
+using Squidex.Domain.Apps.Write.Apps.Commands;
+using Squidex.Infrastructure.Commands;
+using Squidex.Infrastructure.Reflection;
+using Squidex.Pipeline;
+
+namespace Squidex.Areas.Api.Controllers.Apps
+{
+    /// <summary>
+    /// Manages and configures app patterns.
+    /// </summary>
+    [ApiAuthorize]
+    [MustBeAppOwner]
+    [ApiExceptionFilter]
+    [AppApi]
+    [SwaggerTag(nameof(Apps))]
+    public sealed class AppPatternsController : ApiController
+    {
+        public AppPatternsController(ICommandBus commandBus)
+            : base(commandBus)
+        {
+        }
+
+        /// <summary>
+        /// Get app patterns.
+        /// </summary>
+        /// <param name="app">The name of the app.</param>
+        /// <returns>
+        /// 200 => Patterns returned.
+        /// 404 => App not found.
+        /// </returns>
+        /// <remarks>
+        /// Gets all configured regex patterns for the app with the specified name.
+        /// </remarks>
+        [HttpGet]
+        [Route("apps/{app}/patterns/")]
+        [ProducesResponseType(typeof(AppPatternDto[]), 200)]
+        [ApiCosts(1)]
+        public IActionResult GetPatterns(string app)
+        {
+            return Ok(App.Patterns.Values?
```

Comment:
A temporary variable would make debugging much easier.

Reply:
Can we make a fallback here to the config, when the pattern list is null? Would prevent us from having migrations.